### PR TITLE
Cache storage directory

### PR DESCRIPTION
**Pull request by anvox**
Cache directory/bucket at class level to speedup upload/download initialize

### CODE REVIEW

File: lib/tp_common/file_storage/fog_connection.rb [hunk 1]
```diff
@@ -6,13 +6,23 @@ module FileStorage
...
+      def connection
+        @connection ||= Fog::Storage.new(provider: 'AWS',
           aws_access_key_id: FileStorage.configuration.aws_key_id,
           aws_secret_access_key: FileStorage.configuration.aws_secret_key)
       end
+
+      private
+
+      def cached_directories
+        @cached_directories ||= Hash.new do |hash, key|
```
Comment (tung-tinypulse) [inline lib/tp_common/file_storage/fog_connection.rb]: When will this `cached_directories` be refreshed?

Comment (anvox) [reply lib/tp_common/file_storage/fog_connection.rb]: When processes reincarnate :)

**Comment by anvox** [comment]
Tested stages: it could be cached between instances, but not share between classes. 
Improve for whole lib has only 1 cache system.

File: lib/tp_common/file_storage.rb [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
   module FileStorage
-    Configuration = Struct.new(:aws_region,
-                               :aws_key_id,
-                               :aws_secret_key,
-                               :key_prefix,
-                               :default_bucket)
-
     def self.configure
-      @configuration = Configuration.new('us-east-1', nil, nil, nil, nil)
+      @configuration = TpCommon::FileStorage::Configuration.new('us-east-1')
```
Comment (khuyen-nguyen) [inline lib/tp_common/file_storage.rb]: `us-east-1` could we change it to be configurable since we already public this gem.

Comment (anvox) [reply lib/tp_common/file_storage.rb]: Sure, we expose them to configured in initializers. `us-east-1` here is just a default value.

**Pull request merged**